packages/jsii-pacmak/lib/targets/golang/types/class.ts

```diff
@@ -1,29 +1,78 @@
-import { Method, ClassType } from 'jsii-reflect';
+import { Method, ClassType, Initializer } from 'jsii-reflect';
 import { CodeMaker } from 'codemaker';
 import { GoTypeRef } from './go-type-reference';
 import { GoStruct } from './go-type';
 import { TypeField } from './type-field';
+import { getFieldDependencies, substituteReservedWords } from '../util';
 import { Package } from '../package';
+import { ClassConstructor, MethodCall } from '../runtime';
+
+export class GoClassConstructor {
+  private readonly constructorRuntimeCall: ClassConstructor;
+
+  public constructor(
+    public readonly parent: GoClass,
+    private readonly type: Initializer,
+  ) {
+    this.constructorRuntimeCall = new ClassConstructor(this);
+  }
+
+  public emit(code: CodeMaker) {
+    const constr = `New${code.toPascalCase(this.parent.name)}`;
+    const params = this.type.parameters.map(
+      (x) =>
+        `${substituteReservedWords(x.name)} ${new GoTypeRef(
+          this.parent.parent.root,
+          x.type,
+        ).scopedName(this.parent.parent)}`,
```

Comment:
I wonder if we could pull out the return type into a variable so the code is more self-documenting. It'd also make this template string easier to read, so:

```suggestion
        const name = substituteReservedWords(x.name);
        const paramType = new GoTypeRef(this.parent.parent.root,x.type).scopedName(this.parent.parent);
        `${name} ${paramType}`,
```

or something like that.